src/MSBuild/PriorityUtils.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.Diagnostics;
+
+namespace Microsoft.Build.CommandLine
+{
+    internal class PriorityUtils
+    {
+        public static void SwitchProcessPriorityTo(ProcessPriorityClass priority)
```

Comment:
This seems a bit much. Is the catch necessary now?

Reply:
It is if we want to avoid crashing with no helpful message on certain operating systems if a user wants to use this to switch a node to a higher priority, even if that higher priority is changing from low priority to normal priority (which sounds very reasonable). That makes it worth it to me.

Reply:
How would a user switch a node to a higher priority? The only flag we're exposing is low, right?